File: ios/Classes/IASOnboardingsAdaptor.swift
import Foundation
import Flutter

@_spi(QAApp) import InAppStorySDK
@_spi(IAS_API) import InAppStorySDK

class IASOnboardingsAdaptor : IASOnboardingsHostApi {
    init(binaryMessenger: FlutterBinaryMessenger, onboardingsAPI: OnboardingsAPI) {
        self.binaryMessenger = binaryMessenger
        
        self.onboardingsAPI = onboardingsAPI

        self.onboardingLoadCallbackFlutterApi = OnboardingLoadCallbackFlutterApi(binaryMessenger: binaryMessenger);
        
        IASOnboardingsHostApiSetup.setUp(binaryMessenger: binaryMessenger, api: self)
    }
    
    private var binaryMessenger: FlutterBinaryMessenger
    
    private var onboardingsAPI: OnboardingsAPI
    
    private var onboardingLoadCallbackFlutterApi: OnboardingLoadCallbackFlutterApi
    
    func show(limit: Int64, feed: String, tags: [String]) throws {
        func complete(show: Bool) {
            if (show) {
                onboardingLoadCallbackFlutterApi.onboardingLoad(count: limit, feed: feed) { _ in }
            }
        }
        
        onboardingsAPI.showOnboarding(feed: feed, limit: Int(limit), with:tags, with:InAppStory.shared.panelSettings ,complete: complete)
    }
}

File: android/src/main/kotlin/com/example/inappstory_plugin/adaptors/IASOnboardingsAdaptor.kt
package com.example.inappstory_plugin.adaptors

import IASOnboardingsHostApi
import com.example.inappstory_plugin.callbacks.OnboardingLoadCallbackAdaptor
import com.inappstory.sdk.AppearanceManager
import com.inappstory.sdk.core.api.IASOnboardings
import io.flutter.embedding.engine.plugins.FlutterPlugin

class IASOnboardingsAdaptor(
    private val flutterPluginBinding: FlutterPlugin.FlutterPluginBinding,
    private val appearanceManager: AppearanceManager,
    private val iasOnboardings: IASOnboardings,
) : IASOnboardingsHostApi {
    init {
        IASOnboardingsHostApi.setUp(flutterPluginBinding.binaryMessenger, this)

        iasOnboardings.loadCallback(OnboardingLoadCallbackAdaptor(flutterPluginBinding))
    }

    override fun show(limit: Long, feed: String, tags: List<String>) {
        iasOnboardings.show(
            flutterPluginBinding.applicationContext,
            feed,
            appearanceManager,
            tags,
            limit.toInt(),
        )
    }
}
